src/docs/workflows/device-commissioning.mdx

```diff
@@ -0,0 +1,116 @@
+
+### Platform
+
+If a user is doing the device commissioning from mobile phones and tablets, they will have the advantage of a phone camera which can be helpful for scanning QR codes. Other technologies like "BlinkUp" allow device sensors to receive data via your phone's screen blinking. Mobile phones are easier to carry around and can make your physical location less messy, especially if users have instruction manuals and wires flying around everywhere.
+
+On the other hand, computers are better for precise configuration and can process many device details simultaneously due to their superior computing power. The screen size is larger, meaning that you can fit more information and interactions in one screen.
+
+# Design Tricks
+
+## Do not force your users to make decisions on the first go.
+
+If your user does not specify where a device is physically located, will that stop your dashboard from working? Or is it safe to list this device as "locations unspecified" for a while, let the user know that they can skip these settings for now and come back in the future?
+
+## The technician who set it up may not be the final user.
+
+Often, users would invite technicians to come to their home or company and set up the system for them, and later on, the technician would pass the admin right to the user. Sometimes this necessitates a "handoff" experience at the end of your workflow.
+
+## Do not force them to create an account until the need comes up.
+
+Accounts are often used to control the device from anywhere (web, phone, device's HMI) or manage access for different people. However, not all of these functionalities are necessary during an initial device setup. You should only prompt your users to create the account when they actually need it, rather than making the onboarding process burdensome.
+
+## Devices can lose connection halfway through the process.
+
+You should never take for granted your devices will stay connected through the entire process. If there is a network "hiccup," how do you vent users' frustration? Would your application wait & retry in the background, pause the process, or abort the entire workflow?
+
+## It can take a long time before a device responds.
+
+The transmission between IoT devices can take two minutes for jumping between nodes. When your user taps on a button but doesn't see their thermostats working, how would you communicate this latency?
+
+## When a device is installed, its firmware might already be outdated.
+
+There is usually a time gap between when a device is boxed in the factory and when that device gets unboxed by your user. By the time the user plugs it in, your device's firmware might need a firmware upgrade.
+
+Tell your user how long the upgrade might take and whether this will impact the setup process. If the upgrade is non-interruptive, you can have the process running in the background while users are doing the configuration and return to it after they are finished with everything else.
+
```

Comment:
```suggestion
There is usually a time gap between when a device is manufactured and packaged in the factory and when that device gets unboxed by your user. By the time the user plugs it in, your device's firmware might need a firmware upgrade.

Tell your user how long the upgrade might take and whether this will impact the setup process. If the upgrade is non-interruptive, you can have the process running in the background while users are doing the configuration and return to it after they are finished with everything else.

```